server/chat-plugins/help.ts

```diff
@@ -0,0 +1,473 @@
+/**
+ * PS Help room auto-response plugin.
+ * Uses Regex to match room frequently asked question (RFAQ) entries,
+ * and replies if a match is found.
+ * Supports configuration.
+ * Written by mia-pi.
+ */
+
+import {FS} from '../../lib/fs';
+import {Utils} from '../../lib/utils';
+import {LogViewer} from './chatlog';
+import {ROOMFAQ_FILE} from './room-faqs';
+const PATH = 'config/chat-plugins/help.json';
+const MINIMUM_LENGTH = 7;
+const helpRoom = Rooms.get('help');
+const REGEX_WHITELIST = ['miapi', 'annika', 'kris'];
+
+export let helpData: PluginData;
+
+try {
+	helpData = JSON.parse(FS(PATH).readIfExistsSync() || "{}");
+} catch (e) {
+	helpData = {
+		stats: {},
+		pairs: {},
+		disabled: false,
+		queue: [],
+	};
+}
+/**
+ * Terms commonly used in helping that should be ignored
+ * within question parsing (the Help#find function).
+ * */
+const COMMON_TERMS = [
+	'<<(.+)>>', '(do|use|type)( |)(``|)(/|!|//)(.+)(``|)', 'go to', '/rfaq (.+)', 'you can', Chat.linkRegex, 'click',
+	'need to',
+].map(item => new RegExp(item, "i"));
+
+/**
+ * A message filtered by the Help filter.
+ */
+interface LoggedMessage {
+	/** Message that's matched by the Help filter. */
+	message: string;
+	/** The FAQ that it's matched to. */
+	faqName: string;
+	/** The regex that it's matched to. */
+	regex: string;
+}
+/** Object of stats for that day. */
+interface DayStats {
+	matches?: LoggedMessage[];
+	total?: number;
+}
+interface PluginData {
+	/** Stats - filter match and faq that was matched - done day by day. */
+	stats?: {[k: string]: DayStats};
+	/** Word pairs that have been marked as a match for a specific FAQ. */
+	pairs: {[k: string]: string[]};
+	/** Whether or not the filter is disabled. */
+	disabled?: boolean;
+	/** Queue of suggested regex. */
+	queue?: string[];
+}
+
+export class HelpAnswerer {
+	roomFaqs: AnyObject;
+	disabled?: boolean;
+	queue: string[];
+	data: PluginData;
+	constructor(data: PluginData) {
+		this.data = data;
+		this.roomFaqs = this.loadFaqs();
+		this.queue = data.queue ? data.queue : [];
```

Comment:
```suggestion
		this.queue = data.queue || [];
```

(or `?? []`, but PS doesnt like `??`)